File: frontend/add_pointer.py
import re
from pathlib import Path

def patch_file(filepath):
    try:
        with open(filepath, 'r', encoding='utf-8') as f:
            content = f.read()
            
        def button_replacer(match):
            btn_tag = match.group(0)
            if 'cursor-pointer' not in btn_tag:
                if 'className="' in btn_tag:
                    btn_tag = re.sub(r'className="([^"]*)"', r'className="\1 cursor-pointer"', btn_tag)
                else:
                    btn_tag = btn_tag.replace('<button', '<button className="cursor-pointer"')
            return btn_tag
            
        # Only replace inside <button ...>
        final_content = re.sub(r'<button\b[^>]*>', button_replacer, content)
        
        if final_content != content:
            with open(filepath, 'w', encoding='utf-8') as f:
                f.write(final_content)
            print(f"Updated {filepath}")
    except Exception as e:
        print(f"Error on {filepath}: {e}")

src_dir = 'c:/Users/serda/Code/LLM Policy Validator/afstudeerproject-pgm-serdkara/frontend/src'
for filepath in Path(src_dir).rglob("*.tsx"):
    patch_file(filepath)

File: frontend/add_pointer.js
const fs = require('fs');
const path = require('path');

function processDirectory(directory) {
    const files = fs.readdirSync(directory);
    for (const file of files) {
        const fullPath = path.join(directory, file);
        if (fs.statSync(fullPath).isDirectory()) {
            processDirectory(fullPath);
        } else if (fullPath.endsWith('.tsx') || fullPath.endsWith('.jsx')) {
            let content = fs.readFileSync(fullPath, 'utf8');
            let updated = false;

            // Replace <button> tags
            content = content.replace(/<button\b[^>]*>/g, (match) => {
                if (!match.includes('cursor-pointer')) {
                    if (match.includes('className="')) {
                        return match.replace(/className="([^"]*)"/, 'className="$1 cursor-pointer"');
                    } else if (match.includes("className={'")) {
                        return match.replace(/className=\{'([^']*)'\}/, "className={'$1 cursor-pointer'}");
                    } else if (match.includes('className={`')) {
                        return match.replace(/className=\{`([^`]*)`\}/, "className={`$1 cursor-pointer`}");
                    } else {
                        return match.replace('<button', '<button className="cursor-pointer"');
                    }
                }
                return match;
            });

            // Also add cursor-pointer to the known 'Browse Files' div and others if matching
            content = content.replace(/<div[^>]*Browse Files[^<]*<\/div>/g, (match) => {
                // Actually, the 'Browse Files' text is inside the div, my regex on <div ...> won't work well
                return match;
            });

            if (content !== fs.readFileSync(fullPath, 'utf8')) {
                fs.writeFileSync(fullPath, content, 'utf8');
                console.log(`Updated ${fullPath}`);
            }
        }
    }
}

processDirectory(path.join(__dirname, 'src'));
